.test-infra/jenkins/job_beam_Java_Build.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import common_job_properties
+
+// This is the Java Jenkins job which builds artifacts for downstream jobs to consume.
+mavenJob('beam_Java_Build') {
+  description('Builds Java SDK and archives artifacts. Meant to be run as part of a pipeline.')
+
+  // Set standard properties for a job which is part of a pipeline.
+  common_job_properties.setPipelineJobProperties(delegate, 15, "Java Build")
+  // Set standard properties for a pipeline job which needs to pull from GitHub instead of an
+  // upstream job.
+  common_job_properties.setPipelineBuildJobProperties(delegate)
+
+  configure { project ->
+    // The CopyArtifact plugin doesn't support the job DSL so we have to configure it manually.
+    project / 'properties' / 'hudson.plugins.copyartifact.CopyArtifactPermissionProperty' / 'projectNameList' {
+      'string' "beam_*"
+    }
+    // The Build Discarder also doesn't support the job DSL in the right way so we have to configure it manually.
```

Comment:
<!--new_thread; commit:13434ff7764e80f3ff96f6120b9100f537df0f36; resolved:0-->
-1s are "indefinitely/infinite"? So this is "Keep artifacts for a day and other job information forever"?

Reply:
Yes. Added a comment to that effect.